scripts/generate-certs.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+info() {
+  echo "[certs] $*"
+}
+
+error() {
+  echo "[certs] $*" >&2
+}
+
+# Default values
+CERT_DIR="${CERT_DIR:-./artifacts/certs}"
+FORCE="${FORCE:-0}"
+
+# Certificate configuration
+CA_KEY="${CERT_DIR}/ca-key.pem"
+CA_CERT="${CERT_DIR}/ca-cert.pem"
+SERVER_KEY="${CERT_DIR}/spire-server-key.pem"
+SERVER_CERT="${CERT_DIR}/spire-server-cert.pem"
+SERVER_CSR="${CERT_DIR}/spire-server.csr"
+JWT_KEY="${CERT_DIR}/jwt-signing-key.pem"
+BOOTSTRAP_BUNDLE="${CERT_DIR}/bootstrap-bundle.pem"
+
+# Create certs directory if it doesn't exist
+mkdir -p "${CERT_DIR}"
+
+# Check if openssl is available
+if ! command -v openssl >/dev/null 2>&1; then
+  error "openssl is required but not found. Install it and re-run."
+  exit 1
+fi
+
+# Function to check if a file exists and is non-empty
+file_exists() {
+  [[ -f "$1" ]] && [[ -s "$1" ]]
+}
+
+# Function to generate CA certificate and key
+generate_ca() {
+  if [[ "${FORCE}" -eq 0 ]] && file_exists "${CA_KEY}" && file_exists "${CA_CERT}"; then
+    info "CA certificate and key already exist, skipping generation"
+    return 0
+  fi
+
+  info "Generating CA private key (ECDSA P-384)..."
+  openssl ecparam -genkey -name secp384r1 -out "${CA_KEY}"
+
+  info "Generating CA certificate..."
+  openssl req -new -x509 -days 3650 -key "${CA_KEY}" -out "${CA_CERT}" \
+    -subj "/CN=SPIRE CA" \
+    -addext "basicConstraints=critical,CA:TRUE" \
+    -addext "keyUsage=critical,keyCertSign,cRLSign"
+
+  info "CA certificate and key generated successfully"
+}
+
+# Function to generate SPIRE server certificate and key
+generate_spire_server() {
+  if [[ "${FORCE}" -eq 0 ]] && file_exists "${SERVER_KEY}" && file_exists "${SERVER_CERT}"; then
+    info "SPIRE server certificate and key already exist, skipping generation"
+    return 0
+  fi
+
+  info "Generating SPIRE server private key (ECDSA P-256)..."
+  openssl ecparam -genkey -name secp256r1 -out "${SERVER_KEY}"
+
+  info "Generating SPIRE server certificate signing request..."
+  openssl req -new -key "${SERVER_KEY}" -out "${SERVER_CSR}" \
+    -subj "/CN=spire-server"
+
+  info "Generating SPIRE server certificate (signed by CA)..."
+  openssl x509 -req -in "${SERVER_CSR}" -CA "${CA_CERT}" -CAkey "${CA_KEY}" \
+    -CAcreateserial -out "${SERVER_CERT}" -days 365 \
+    -extensions v3_server -extfile <(
+      echo "[v3_server]"
+      echo "basicConstraints=CA:FALSE"
+      echo "keyUsage=digitalSignature"
+      echo "extendedKeyUsage=serverAuth"
+      echo "subjectAltName=DNS:spire-server,DNS:spire-server.default.svc.cluster.local"
+    )
+
+  # Clean up CSR file
+  rm -f "${SERVER_CSR}"
+
+  info "SPIRE server certificate and key generated successfully"
+}
+
+# Function to generate JWT signing key
+generate_jwt_key() {
```

Comment:
The JWT signing key was included as part of the original issue requirements (#8), which specified generating 'JWT signing key' along with the CA, SPIRE server cert/key, and bootstrap bundle. 

However, you raise a good point - if SPIRE doesn't actually require a separate JWT signing key for this use case, we could remove it to simplify. SPIRE typically uses its own internal key management for JWT tokens.

Should I remove the JWT key generation, or is there a specific use case where it's needed?

Reply:
please remove it